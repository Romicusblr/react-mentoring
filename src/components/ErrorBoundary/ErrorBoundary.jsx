import React from 'react';
import PropTypes from 'prop-types';

class ErrorBoundary extends React.Component {
  constructor(props) {
    super(props);
    this.state = { hasError: false };
  }

  componentDidCatch() {
    this.setState({
      hasError: true,
    });
  }

  render() {
    const { hasError } = this.state;
    const { children } = this.props;
    if (hasError) {
      return <h2>Something went wrong.</h2>;
    }
    return children;
  }
}

ErrorBoundary.propTypes = {
  children: PropTypes.node.isRequired,
};

export default ErrorBoundary;
