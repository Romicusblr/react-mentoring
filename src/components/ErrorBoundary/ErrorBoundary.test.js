import ErrorBoundary from '.';

const Something = () => <h1>Foo</h1>;

describe('ErrorBoundary', () => {
  it('should match to snapshot without error', () => {
    const snap = shallow(<ErrorBoundary><Something /></ErrorBoundary>);
    expect(snap).toMatchSnapshot();
  });

  it('should match to snapshot with error', () => {
    const snap = shallow(<ErrorBoundary><Something /></ErrorBoundary>);
    snap.find(Something).simulateError(new Error());
    expect(snap).toMatchSnapshot();
  });
});
