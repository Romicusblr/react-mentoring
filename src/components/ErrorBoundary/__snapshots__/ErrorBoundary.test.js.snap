// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary should match to snapshot with error 1`] = `
<h2>
  Something went wrong.
</h2>
`;

exports[`ErrorBoundary should match to snapshot without error 1`] = `<Something />`;
